code/game/objects/structures/ai_core.dm

```diff
@@ -25,218 +31,190 @@
-				user << "<span class='notice'>You start to deconstruct the frame...</span>"
-				if(do_after(user, 20/P.toolspeed, target = src))
-					if(!src || !WT.remove_fuel(0, user)) return
-					user << "<span class='notice'>You deconstruct the frame.</span>"
-					new /obj/item/stack/sheet/plasteel( loc, 4)
-					qdel(src)
-				return
-		if(1)
-			if(istype(P, /obj/item/weapon/wrench))
-				playsound(loc, 'sound/items/Ratchet.ogg', 50, 1)
-				user << "<span class='notice'>You start to unfasten the frame...</span>"
-				if(do_after(user, 20/P.toolspeed, target = src))
-					user << "<span class='notice'>You unfasten the frame.</span>"
-					anchored = 0
-					state = 0
-				return
-			if(istype(P, /obj/item/weapon/circuitboard/aicore) && !circuit)
-				if(!user.drop_item())
-					return
-				playsound(loc, 'sound/items/Deconstruct.ogg', 50, 1)
-				user << "<span class='notice'>You place the circuit board inside the frame.</span>"
-				icon_state = "1"
-				circuit = P
-				P.loc = src
-				return
-			if(istype(P, /obj/item/weapon/screwdriver) && circuit)
-				playsound(loc, 'sound/items/Screwdriver.ogg', 50, 1)
-				user << "<span class='notice'>You screw the circuit board into place.</span>"
-				state = 2
-				icon_state = "2"
-				return
-			if(istype(P, /obj/item/weapon/crowbar) && circuit)
-				playsound(loc, 'sound/items/Crowbar.ogg', 50, 1)
-				user << "<span class='notice'>You remove the circuit board.</span>"
-				state = 1
-				icon_state = "0"
-				circuit.loc = loc
-				circuit = null
+	if(istype(P, /obj/item/weapon/wrench))
+		playsound(loc, 'sound/items/Ratchet.ogg', 50, 1)
+		user.visible_message("[user] [anchored ? "fastens" : "unfastens"] [src].", \
+					 "<span class='notice'>You start to [anchored ? "unfasten [src] from" : "fasten [src] to"] the floor...</span>")
+		if(do_after(user, 20/P.toolspeed, target = src))
+			user << "<span class='notice'>You [anchored ? "unfasten [src] from" : "fasten [src] to"] the floor.</span>"
+			anchored = !anchored
+		return
+	if(!anchored)
+		if(istype(P, /obj/item/weapon/weldingtool))
+			if(state != EMPTY_CORE)
+				user << "<span class='warning'>The core must be empty to deconstruct it!</span>"
 				return
-		if(2)
-			if(istype(P, /obj/item/weapon/screwdriver) && circuit)
-				playsound(loc, 'sound/items/Screwdriver.ogg', 50, 1)
-				user << "<span class='notice'>You unfasten the circuit board.</span>"
-				state = 1
-				icon_state = "1"
+			var/obj/item/weapon/weldingtool/WT = P
+			if(!WT.isOn())
+				user << "<span class='warning'>The welder must be on for this task!</span>"
 				return
-			if(istype(P, /obj/item/stack/cable_coil))
-				var/obj/item/stack/cable_coil/C = P
-				if(C.get_amount() >= 5)
+			playsound(loc, 'sound/items/Welder.ogg', 50, 1)
+			user << "<span class='notice'>You start to deconstruct the frame...</span>"
+			if(do_after(user, 20/P.toolspeed, target = src) && src && state == EMPTY_CORE && WT && WT.remove_fuel(0, user))

```

Comment:
the src check is useless, what you need is !qdeleted(src) (or if you're cheap and feel lucky "if(loc)" )